File: CouponsProject/src/app/models/coupon.ts
import { CouponCategory } from './../enums/coupon-category.enum';
export class Coupon {
    // credit to TypeScript ToolBox for auto generating 90% of this (lazy = efficent ;) )

    private couponId : number;
    private amount : number;
    private price : number;
    private title : string;
    private description : string;
    private image : string;
    private startDate : Date;
    private endDate : Date;
    private couponCategory : CouponCategory;

	constructor($couponId: number, $amount: number, $price: number, $title: string, $description: string, $image: string, $startDate: Date, $endDate: Date, $couponCategory: CouponCategory) {
		this.couponId = $couponId;
		this.amount = $amount;
		this.price = $price;
		this.title = $title;
		this.description = $description;
		this.image = $image;
		this.startDate = $startDate;
		this.endDate = $endDate;
		this.couponCategory = $couponCategory;
	}

    /**
     * Getter $couponId
     * @return {number}
     */
	public get $couponId(): number {
		return this.couponId;
	}

    /**
     * Getter $amount
     * @return {number}
     */
	public get $amount(): number {
		return this.amount;
	}

    /**
     * Getter $price
     * @return {number}
     */
	public get $price(): number {
		return this.price;
	}

    /**
     * Getter $title
     * @return {string}
     */
	public get $title(): string {
		return this.title;
	}

    /**
     * Getter $description
     * @return {string}
     */
	public get $description(): string {
		return this.description;
	}

    /**
     * Getter $image
     * @return {string}
     */
	public get $image(): string {
		return this.image;
	}

    /**
     * Getter $startDate
     * @return {Date}
     */
	public get $startDate(): Date {
		return this.startDate;
	}

    /**
     * Getter $endDate
     * @return {Date}
     */
	public get $endDate(): Date {
		return this.endDate;
	}

    /**
     * Getter $couponCategory
     * @return {CouponCategory}
     */
	public get $couponCategory(): CouponCategory {
		return this.couponCategory;
	}

    /**
     * Setter $couponId
     * @param {number} value
     */
	public set $couponId(value: number) {
		this.couponId = value;
	}

    /**
     * Setter $amount
     * @param {number} value
     */
	public set $amount(value: number) {
		this.amount = value;
	}

    /**
     * Setter $price
     * @param {number} value
     */
	public set $price(value: number) {
		this.price = value;
	}

    /**
     * Setter $title
     * @param {string} value
     */
	public set $title(value: string) {
		this.title = value;
	}

    /**
     * Setter $description
     * @param {string} value
     */
	public set $description(value: string) {
		this.description = value;
	}

    /**
     * Setter $image
     * @param {string} value
     */
	public set $image(value: string) {
		this.image = value;
	}

    /**
     * Setter $startDate
     * @param {Date} value
     */
	public set $startDate(value: Date) {
		this.startDate = value;
	}

    /**
     * Setter $endDate
     * @param {Date} value
     */
	public set $endDate(value: Date) {
		this.endDate = value;
	}

    /**
     * Setter $couponCategory
     * @param {CouponCategory} value
     */
	public set $couponCategory(value: CouponCategory) {
		this.couponCategory = value;
	}

}

File: Old/JB/Project33_lesson_And_bigProject_phase1_27.01.2020/src/big_project/beans/Coupon.java
package big_project.beans;

import java.sql.Date;

public class Coupon {
	private int couponId, companyId, customerId;
	private int amount, price;
	private String title, description, image;
	private Date startDate, endDate;
	private CategoryType category;

	public Coupon(int couponId, int companyId, int customerId, int amount, int price, String title, String description,
			String image, Date startDate, Date endDate, CategoryType category) {
		this.couponId = couponId;
		this.companyId = companyId;
		this.customerId = customerId;
		this.amount = amount;
		this.price = price;
		this.title = title;
		this.description = description;
		this.image = image;
		this.startDate = startDate;
		this.endDate = endDate;
		this.category = category;
	}

	public Coupon(int companyId, int customerId, int amount, int price, String title, String description, String image,
			Date startDate, Date endDate, CategoryType category) {
		this.companyId = companyId;
		this.customerId = customerId;
		this.amount = amount;
		this.price = price;
		this.title = title;
		this.description = description;
		this.image = image;
		this.startDate = startDate;
		this.endDate = endDate;
		this.category = category;
	}

	public int getCompanyId() {
		return companyId;
	}

	public void setCompanyId(int companyId) {
		this.companyId = companyId;
	}

	public int getCustomerId() {
		return customerId;
	}

	public void setCustomerId(int customerId) {
		this.customerId = customerId;
	}

	public int getAmount() {
		return amount;
	}

	public void setAmount(int amount) {
		this.amount = amount;
	}

	public int getPrice() {
		return price;
	}

	public void setPrice(int price) {
		this.price = price;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getImage() {
		return image;
	}

	public void setImage(String image) {
		this.image = image;
	}

	public Date getStartDate() {
		return startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public Date getEndDate() {
		return endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public int getCouponId() {
		return couponId;
	}

	public CategoryType getCategory() {
		return category;
	}

	public void setCategory(CategoryType category) {
		this.category = category;
	}

	@Override
	public String toString() {
		return "Coupon [couponId=" + couponId + ", companyId=" + companyId + ", customerId=" + customerId + ", amount="
				+ amount + ", price=" + price + ", title=" + title + ", description=" + description + ", image=" + image
				+ ", startDate=" + startDate + ", endDate=" + endDate + ", category=" + category + "]";
	}

}
